src/lib/components/Swap/RoutingDiagram/index.tsx

```diff
@@ -0,0 +1,139 @@
+  padding: 0.1em 0.5em;
+  position: relative;
+`
+
+const DottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  position: absolute;
+  width: calc(100% - 1em);
+  z-index: ${Layer.UNDERLAYER};
+`
+
+const ShortDottedLine = styled(DottedLine)`
+  overflow: hidden;
+  position: relative;
+  width: 71px;
+`
+
+const DotColor = styled(DotLine)`
+  path {
+    stroke: ${({ theme }) => theme.secondary};
+  }
+`
+
+const DetailsRow = styled(Row)`
+  display: grid;
+  grid-template-columns: 4.8125em 1fr;
+  width: 100%;
+`
+
+const StyledAutoRouterLabel = styled(ThemedText.ButtonSmall)`
+  color: ${({ theme }) => theme.primary};
+  font-weight: 500;
+  line-height: 1em;
+
+  @supports (-webkit-background-clip: text) and (-webkit-text-fill-color: transparent) {
+    background-image: linear-gradient(90deg, #2172e5 0%, #54e521 163.16%);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+  }
+`
+
+const StyledAutoRouterIcon = styled(AutoRouterIcon)`
+  height: 0.875em;
+  width: 0.875em;
+`
+
+function Pool({ currency0, currency1, feeAmount }: { currency0: Currency; currency1: Currency; feeAmount: FeeAmount }) {
+  return (
+    <Badge padding="0 4px" color="dialog">
+      <Badge gap={0.375}>
+        <Row>
+          <TokenImg token={currency0} />
+          <TokenImg token={currency1} />
+        </Row>
+        <ThemedText.Subhead2>{feeAmount / 10000}%</ThemedText.Subhead2>
+      </Badge>
+    </Badge>
+  )
+}
+
+export default function RoutingDiagram({ trade }: { trade: InterfaceTrade<Currency, Currency, TradeType> }) {
+  const routes: RoutingDiagramEntry[] = useMemo(() => getTokenPath(trade), [trade])
+
+  return (
+    <Wrapper gap={0.75}>
+      <Row justify="space-between">
+        <Row gap={0.25}>
+          <StyledAutoRouterIcon />
+          <StyledAutoRouterLabel>
+            <Trans>Auto Router</Trans>
+          </StyledAutoRouterLabel>
+        </Row>
+        <ThemedText.ButtonSmall>
+          <Trans>
+            Best route via {routes.length} <Plural value={routes.length} one="hop" other="hops" />
+          </Trans>
+        </ThemedText.ButtonSmall>
+      </Row>
+      <Rule />
+      {routes.map((route, index) => (
+        <RouteRow key={index} align="center">
+          <TokenImg token={trade.inputAmount.currency} />
+          <ShortDottedLine>
+            <DotColor />
+          </ShortDottedLine>
+          <RouteDetailsContainer justify="flex-start" flex>
+            <DottedLine>
+              <DotColor />
+            </DottedLine>
+            <DetailsRow>
+              <Badge padding="0 4px" color="dialog">
+                <Badge gap={0.375}>
+                  <ThemedText.ButtonSmall color="secondary">{route.percent.toSignificant(2)}%</ThemedText.ButtonSmall>
+                  <Badge padding="0.125em" borderRadius={0.25} color="module">
+                    <ThemedText.Badge color="secondary">{route.protocol.toUpperCase()}</ThemedText.Badge>
```

Comment:
This is not sized correctly in the vercel preview.

Reply:
im wondering if the classname "badge" is causing this weird behvaior, for now i can fix with setting the size explicilty